### DISCUSSION TITLE
Bookkeeper Storage recommendations

### DISCUSSION THREAD

**Discussion by Apurva007** [apache/pulsar#19606] [Q&A] [answered]
Hello, 

My team is currently exploring Apache Pulsar. We had a few questions on the storage configuration of the bookkeeper.
1. What is the recommended storage ratio of journal to the ledger disks? We have 6 ledger SSD SATA disks of 7.5 TB each available. We were wondering how much journal storage(NVMe) would be required?
2. Which file system is recommended for Pulsar? Kafka recommends utilizing xfs filesystem. Was wondering if its the same for pulsar as well?

Any help would be highly appreciated.

**Comment by hangc0276** [comment, marked as answer]
> What is the recommended storage ratio of journal to the ledger disks? We have 6 ledger SSD SATA disks of 7.5 TB each available. We were wondering how much journal storage(NVMe) would be required?

The journal disk doesn't have the storage size requirement, maybe 100GB is enough. You need to ensure the journal throughput can match with your 6 ledger SSD throughput. 

> Which file system is recommended for Pulsar? Kafka recommends utilizing xfs filesystem. Was wondering if its the same for pulsar as well?

Both XFS and ext4 can be good options to use with Apache Bookkeeper. If performance is a top priority and you have large segments and high-throughput workloads, XFS may be the better choice. If reliability and stability are more important, then ext4 may be the better choice.

**Reply by Apurva007** [reply]
@hangc0276 Great! thanks for helping with the answer! 😄
